broker/nats.go

```diff
@@ -64,9 +65,9 @@ func (b broker) Publish(_ context.Context, _ string, msg Message) error {
 		return err
 	}
 
-	subject := fmt.Sprintf("%s.%s", prefix, msg.Channel)
+	subject := fmt.Sprintf("%s.%s", chansPrefix, msg.Channel)
 	if msg.Subtopic != "" {
-		subject = fmt.Sprintf("%s.%s", subject, msg.Subtopic)
+		subject = fmt.Sprintf("%s.%s", chansPrefix, msg.Subtopic)
```

Comment:
Looks like we are changing the logic - we used previously defined `subject` (on the line above), now we are using chan prefix, and this looks incorrect - as subtopic is actually added to a topic, not to a prefix.